packages/object-extend/index.js

```diff
@@ -70,3 +79,13 @@ function isCloneable(obj) {
 function isUnextendable(val) {
   return !val || (typeof val != 'object' && typeof val != 'function');
 }
+function hasOwnPropertySafe(obj, key) {
+  try {
+    return Object.prototype.hasOwnProperty.call(obj, key);
+  } catch (e) {
+    if (e instanceof TypeError) {
+      return true;
+    }
+    throw e;
+  }
+}
```

Comment:
I would not bother with the try/catch here. The aim of Just is to be minimally sized and not to cover extremely unlikely edge cases (as Lodash would). If Object.prototype is polluted then honestly all bets are off for the codebase.